pwndbg/aglib/heap/ptmalloc.py

```diff
@@ -1250,6 +1250,16 @@ def tcachebins(self, tcache_addr: int | None = None) -> Bins | None:
         if tcache is None:
             return None
 
+        global tcache_2_42_warning_issued
+        if pwndbg.glibc.get_version() >= (2, 42) and "tcache_2_42_warning_issued" not in globals():
```

Comment:
```suggestion
        if pwndbg.glibc.get_version() >= (2, 42) and tcache_2_42_warning_issued is False:
```
and `tcache_2_42_warning_issued = False` must be implemented globally outside this func/class

Reply:
this won't work because it isn't initialized at that point in time, e.g. try this example:
```python
def print_once():
    global once
    if once is False:
        print("once")
        once = True
```
will throw `NameError: name 'once' is not defined`, but this works
```python
def print_once():
    global once
    if 'once' not in globals():
        print("once")
        once = True
```
i know it is a hacky fix but it works for this use cases

Reply:
ok this works but review dog says no
```
  Name "tcache_2_42_warning_issued" is not defined  [name-defined]
```

Reply:
any other ideas for one time warning

Reply:
Heh, this would work if u set the global variable in global scope :D.

Reply:
now using this, it works but let's see what review dog says:
```
        if pwndbg.glibc.get_version() >= (2, 42) and not hasattr(
            self, "tcache_2_42_warning_issued"
        ):
            print(
                message.warn(
                    "Changes to tcache in GLIBC 2.42 have not been fully implemented. "
                    "PR contributions are highly appreciated!"
                )
            )
            setattr(self, "tcache_2_42_warning_issued", True)
```

Reply:
initially wanted to do sth like this
```python
def print_once():
    if not hasattr(print_once, "once"):
        print("once")
        setattr(print_once, "once", True)
```
but not sure what the equiv for object methods would be

Reply:
nvmd i can literally just do that